packages/web/components/Modals/UnsplashModal/UnsplashModal.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import { createApi } from 'unsplash-js'
+
+import Modal from '@/components/Modal'
+import SearchInput from '@/components/Dashboard/Filters/SearchInput'
+
+type Props = {
+  onCancel: () => void
+  onUnsplashSelect: (smallUrl: string, largeUrl: string) => void
+  show: boolean
+}
+
+type Photo = {
+  id: number
+  width: number
+  height: number
+  urls: { full: string; regular: string; raw: string; small: string; thumb: string }
+  color: string | null
+  user: {
+    username: string
+    name: string
+  }
+}
+
+const unsplashApi = createApi({
+  accessKey: process.env.NEXT_PUBLIC_UNSPLASH_ACCESS_KEY as string,
+})
+
+const PhotoComp: React.FC<{ photo: Photo; onImageSelect: any }> = ({ photo, onImageSelect }) => {
+  const { user, urls } = photo
+
+  return (
+    <>
+      <img
+        className="img"
+        src={urls.thumb}
+        onClick={() => onImageSelect(urls.small, urls.regular)}
+      />
+      <a className="credit" target="_blank" href={`https://unsplash.com/@${user.username}`}>
+        {user.name}
+      </a>
+      <style jsx>{`
+        .img {
+          position: relative;
+          margin-bottom: 4px;
+          width: 100%;
+          max-height: 300px;
+          object-fit: cover;
+        }
+
+        .img:hover {
+          cursor: pointer;
+        }
+
+        .credit {
+          font-size: 12px;
+          line-height: 15px;
+          text-decoration-line: underline;
+          color: #b3b3b3;
+        }
+      `}</style>
+    </>
+  )
+}
+
+const UnsplashSearch: React.FC<{ onImageSelect: any }> = ({ onImageSelect }) => {
+  const [data, setPhotosResponse] = React.useState<any>()
```

Comment:
unsplash-js is a typescript project so I think it should come with some type declarations we can use instead of `any` here and the custom photo type above. Even if they don't export the types we need directly (which it kinda looks like they don't, and that's regrettable) we can do something like:

```
type Awaited<T> = T extends PromiseLike<infer U> ? Awaited<U> : T
type SearchResponse = Awaited<ReturnType<ReturnType<typeof createApi>['search']['getPhotos']>>
```

and use `SearchResponse` as our type here